### ISSUE TITLE
Allow for a11yTitle to be passed into List

### ISSUE THREAD

**Issue by amandadupell** [grommet/grommet#6047]
### Expected Behavior
When passing in an `a11yTitle` for the List component, there should be an `aria-label` applied to the List.

### Actual Behavior
When passing in an `a11yTitle` for the List component, there is no `aria-label`.

### URL, screen shot, or Codepen exhibiting the issue
[Code](https://codesandbox.io/s/grommet-v2-template-forked-m5zhee?file=/index.js)

### Steps to Reproduce

1. Open example
2. See there is an `a11yTitle` for List
3. Test with screen reader. `a11yTitle` is not read by screen reader and there is no `aria-label` on the `<ul/>`.

### Your Environment

- Grommet version: v2.22.0
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile):
